### ISSUE TITLE
Better Fraction Support

### ISSUE THREAD

**Issue by 2426021684** [nlp-compromise/nlp_compromise#195]
- [ ] Handle Spaces
- [ ] Vulgar Fractions (Single Code Point)
- [ ] Subscript, Superscripts, and Unicode Fractions Slash
- [ ] Handle words

**Comment by 2426021684** [comment]
Should I replace the fancy unicode characters with the normals before parsing or should I handle them during parsing?

**Comment by spencermountain** [comment]
good question. i'd support your choice. they're already tagged as `Value` in the pos-tagger now right?
it would make sense to me that they be passed-around as unicode, and only become an integer after value.parse() runs
¯\_(ツ)_/¯

**Comment by 2426021684** [comment]
They are not tagged as `Value`

**Comment by 2426021684** [comment]
Put them in data/number.js right?

**Comment by spencermountain** [comment]
yeah, then in `./src/lexicon.js`
thanks.

**Comment by 2426021684** [comment]
How can I ensure that when parsing "1½" that "1" and "½" are parsed separately (in the to_number method)?

**Comment by spencermountain** [comment]
yeah, if tagger.js identifies them both as `Value`, they'll be lumped together by `lumper.js`.
so that to_number method will need to split them back apart, which i think you're already doing, if i'm correct.

... i guess if a fraction appears before a number, it's `½ * num`, but if it appears after, it's `num + ½`?

i think the decimal logic in to_number is similar - eg '"two point eight nine"`. Dunno.
it may be sensible to split-out `fraction_parse, decimal_parse, etc` in to_number, to make your life more pleasant. Up to you. :bow:

**Comment by 2426021684** [comment]
"-1½" is parsing. For some reason "- 1" fails.

**Comment by 2426021684** [comment]
"- 1" working

**Comment by 2426021684** [comment]
"- 1 ½ 1/2" is -2. Should it be different?

**Comment by 2426021684** [comment]
For superscripts should 3³ be 33 or 9 or nothing?

**Comment by spencermountain** [comment]
hey @2426021684 is this ready to pull in?

**Comment by 2426021684** [comment]
Sorry I forgot about this. I still have to figure out how to get fractions recognized as values.
